### PR TITLE
Create preview PRs using labels

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

Benefits:

1. Ease of use - straight from the PR page
2. Workflow runs will show on the PR checks¹
3. Downstream PRs will be automatically updated with new PR changes
4. Simplified logic in reusable workflow

¹ In the case that a preview label already exists and another preview
label is added, this results in the workflow for the label that already
exists to appear as "skipped" because it has been skipped on the latest
event trigger to avoid an unnecessary update.

## Checklist

- [x] Checks pass, though they don't test the changes here
- [x] ~If making user-facing changes, add a message in [CHANGELOG.md](https://github.com/nextstrain/auspice/blob/HEAD/CHANGELOG.md) summarizing the changes in this PR~ no functional changes
- [x] (to be done by a Nextstrain team member) [Create preview PRs on downstream repositories][1].
- [x] PR previews are triggered on label
- [x] PR previews are auto-updated on new PR changes
- [x] Workflow runs show under PR checks
- [x] Dev docs updated

[1]: https://github.com/nextstrain/auspice/blob/-/DEV_DOCS.md#test-on-downstream-repositories

### CODE REVIEW

File: .github/workflows/preview_on_downstream_repo.yaml [hunk 1]
```diff
@@ -9,42 +9,19 @@ on:
         required: false
         type: string
         default: .
-jobs:
-  make-pr-on-downstream-repo:
-    # I don't see this being used for tags, so ensure it's only run on branches
-    # to make subsequent logic and wording easier.
-    if: github.ref_type == 'branch'
 
+concurrency:
```
[the remark this answers is not in the document]
Comment (victorlin) [reply .github/workflows/preview_on_downstream_repo.yaml:13]: Yes I copied from that example! I searched briefly and couldn't find docs, so I just did some testing to make sure that concurrency was per caller workflow, not the reusable workflow itself.

I'll add a comment here so this is more clear.

**Comment by victorlin** [comment]
> I haven't seen this approach used before -- did you see it elsewhere or come up with it yourself?

The idea popped in my head while working on #1765. I searched online for "trigger on label" which led me to a [StackOverflow example](https://stackoverflow.com/a/62331521/4410590), and I took it from there.

File: .github/workflows/preview_on_downstream_repo.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (victorlin) [reply .github/workflows/preview_on_downstream_repo.yaml:14]: Comment added in [force-push](https://github.com/nextstrain/auspice/compare/a0d2ef36e62bf984367e694459f7888aed4149f7..90b7dee4627d1ec07a2e3f6d782b661804302991).

**Comment by victorlin** [comment]
I'll merge this now since this is a change on the dev side that we can iterate on.

File: .github/workflows/preview_on_auspice_us.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      - labeled
+
+  workflow_dispatch:
+
+jobs:
+  run:
+    # Run if PR has the label and is updated or is newly labeled with the label
+    if: |
+      (github.event.action != 'labeled' && contains(github.event.pull_request.labels.*.name, 'preview on auspice.us')) ||
+      (github.event.action == 'labeled' && github.event.label.name == 'preview on auspice.us')
```
Comment (joverlee521) [inline .github/workflows/preview_on_auspice_us.yaml:17]: I believe this logic prevents runs from `workflow_dispatch` events, should we just remove the `workflow_dispatch` trigger? (I just tried a [manual run on master](https://github.com/nextstrain/auspice/actions/runs/8694430624) and the job was skipped).

Comment (victorlin) [reply .github/workflows/preview_on_auspice_us.yaml:17]: Oh yes, not sure why I copied it. Removing in #1768

**Comment by tsibley** [comment]
Did anyone ensure that labels can only be added by us and not random other GitHub users?

**Comment by victorlin** [comment]
That's a good concern, it hadn't crossed my mind. But I don't think anyone can add labels... I just tried on a random PR I had opened in [bioconda-recipes](https://github.com/bioconda/bioconda-recipes/pulls/@me) and don't see a way to add a label other than tagging `@BiocondaBot please add label` which is something they had to strictly allow.

From [GitHub docs](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#applying-a-label):

> Anyone with triage access to a repository can apply and dismiss labels.

**Comment by tsibley** [comment]
Yay, that's great. I'm not sure why I recalled being able to add labels to issues/PRs I've created in third-party repos…

**Pull request merged**